Sources/Build/BuildOperationBuildSystemDelegateHandler.swift

```diff
@@ -214,34 +211,53 @@ final class TestEntryPointCommand: CustomLLBuildCommand, TestBuildCommand {
             throw InternalError("main file output (\(TestEntryPointTool.mainFileName)) not found")
         }
 
-        let testObservabilitySetup: String
-        if self.context.buildParameters.testingParameters.experimentalTestOutput
-            && self.context.buildParameters.targetTriple.supportsTestSummary {
-            testObservabilitySetup = "_ = SwiftPMXCTestObserver()\n"
-        } else {
-            testObservabilitySetup = ""
-        }
-
         // Write the main file.
         let stream = try LocalFileOutputByteStream(mainFile)
 
-        stream.send(
-            #"""
-            \#(generateTestObservationCode(buildParameters: self.context.buildParameters))
+        if self.context.buildParameters.testingParameters.library == .swiftTesting {
+            stream.send(
+                #"""
+                #if canImport(Testing)
+                @_spi(SwiftPackageManagerSupport) import Testing
+                @main struct Runner {
+                    static func main() async {
+                        await Testing.swiftPMEntryPoint() as Never
+                    }
+                }
+                #endif
+                """#
+            )
+        } else {
+            // Find the inputs, which are the names of the test discovery module(s)
+            let inputs = tool.inputs.compactMap { try? AbsolutePath(validating: $0.name) }
```

Comment:
loosing error information? should thee be paths all the way?

Reply:
The existing code also uses `try?` but I can change it to throwing if we think it's not failable.

Reply:
I'm going to leave this as-is because it's the existing logic and I don't know the code well enough to assert it cannot ever throw.